### PR TITLE
Fix train_loader not initalized properly

### PR DESCRIPTION
**Pull request by Louis-Dupont**
I am not sure why this error happens to begin (see https://github.com/Deci-AI/super-gradients/issues/967), and I couldnt reproduce it, but this should fix it:
- Make the code more robust to this type of error
- Raise a more explicit error when user doesnt provide a loader and not loader found in cache.

### CODE REVIEW

**Review by shaydeci — APPROVED** [review]
LGTM
